```
uv-lock..................................................................Passed
ruff (legacy alias)......................................................Failed
- hook id: ruff
- exit code: 1

tests/bedrock_agentcore/memory/integrations/strands/test_agentcore_memory_session_manager.py:3262:121: E501 Line too long (122 > 120)
     |
3261 |     def test_config_accepts_none_mode(self):
3262 |         config = AgentCoreMemoryConfig(memory_id="m", session_id="s", actor_id="a", persistence_mode=PersistenceMode.NONE)
     |                                                                                                                         ^^ E501
3263 |         assert config.persistence_mode is PersistenceMode.NONE
     |

tests/bedrock_agentcore/memory/integrations/strands/test_agentcore_memory_session_manager.py:3373:121: E501 Line too long (122 > 120)
     |
3371 |     def test_read_agent_works(self, no_persist_manager, mock_memory_client):
3372 |         mock_memory_client.list_events.return_value = [
3373 |             {"eventId": "e1", "payload": [{"blob": '{"agent_id": "a1", "state": {}, "conversation_manager_state": {}}'}]},
     |                                                                                                                         ^^ E501
3374 |         ]
3375 |         result = no_persist_manager.read_agent("test-session-456", "a1")
     |

tests/bedrock_agentcore/memory/integrations/strands/test_agentcore_memory_session_manager.py:3384:121: E501 Line too long (169 > 120)
     |
3382 | …
3383 | …
3384 | …"role": "user", "content": [{"text": "Hello"}]}, "message_id": 1}'}, "role": "USER"}}],
     |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E501
3385 | …
3386 | …
     |

tests/bedrock_agentcore/memory/integrations/strands/test_agentcore_memory_session_manager.py:3395:121: E501 Line too long (124 > 120)
     |
3393 |         mock_memory_client.list_events.side_effect = [
3394 |             [],
3395 |             [{"eventId": "legacy-1", "payload": [{"blob": '{"session_id": "test-session-456", "session_type": "AGENT"}'}]}],
     |                                                                                                                         ^^^^ E501
3396 |         ]
3397 |         result = no_persist_manager.read_session("test-session-456")
     |

tests/bedrock_agentcore/memory/integrations/strands/test_agentcore_memory_session_manager.py:3405:121: E501 Line too long (130 > 120)
     |
3403 |         mock_memory_client.list_events.side_effect = [
3404 |             [],
3405 |             [{"eventId": "legacy-1", "payload": [{"blob": '{"agent_id": "a1", "state": {}, "conversation_manager_state": {}}'}]}],
     |                                                                                                                         ^^^^^^^^^^ E501
3406 |         ]
3407 |         result = no_persist_manager.read_agent("test-session-456", "a1")
     |

Found 5 errors.

ruff format..............................................................Failed
- hook id: ruff-format
- files were modified by this hook

2 files reformatted, 152 files left unchanged

trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check toml...............................................................Passed
check json...............................................................Passed
check yaml...............................................................Passed
check for merge conflicts................................................Passed
check for added large files..............................................Passed
debug statements (python)................................................Passed
```

small lint errors to address before merging.

--- a/src/bedrock_agentcore/memory/integrations/strands/session_manager.py
+++ b/src/bedrock_agentcore/memory/integrations/strands/session_manager.py
@@ -142,7 +142,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
         """
         self.converter = converter or AgentCoreMemoryConverter
         self.config = agentcore_memory_config
-        self.read_only = agentcore_memory_config.persistence_mode is PersistenceMode.NONE
+        self.persistence_mode = agentcore_memory_config.persistence_mode
         self.memory_client = MemoryClient(region_name=region_name)
         session = boto_session or boto3.Session(region_name=region_name)
         self.has_existing_agent = False
@@ -256,7 +256,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
         if session.session_id != self.config.session_id:
             raise SessionException(f"Session ID mismatch: expected {self.config.session_id}, got {session.session_id}")
 
-        if not self.read_only:
+        if self.persistence_mode is not PersistenceMode.NONE:
             event = self.memory_client.gmdp_client.create_event(
                 memoryId=self.config.memory_id,
                 actorId=self.config.actor_id,
@@ -267,9 +267,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
                 eventTimestamp=self._get_monotonic_timestamp(),
                 metadata={STATE_TYPE_KEY: {"stringValue": StateType.SESSION.value}},
             )
-            logger.info(
-                "Created session: %s with event: %s", session.session_id, event.get("event", {}).get("eventId")
-            )
+            logger.info("Created session: %s with event: %s", session.session_id, event.get("event", {}).get("eventId"))
 
         return session
 
@@ -323,7 +321,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
             session_data = json.loads(old_event.get("payload", {})[0].get("blob"))
             session = Session.from_dict(session_data)
             # Migrate: create new event with metadata, delete old
-            if not self.read_only:
+            if self.persistence_mode is not PersistenceMode.NONE:
                 self.create_session(session)
                 self.memory_client.gmdp_client.delete_event(
                     memoryId=self.config.memory_id,
@@ -370,7 +368,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
         if session_agent.created_at:
             self._agent_created_at_cache[session_agent.agent_id] = session_agent.created_at
 
-        if self.read_only:
+        if self.persistence_mode is PersistenceMode.NONE:
             return
 
         if self.config.batch_size > 1:
@@ -471,7 +469,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
                 agent_data = json.loads(old_event.get("payload", {})[0].get("blob"))
                 agent = SessionAgent.from_dict(agent_data)
                 # Migrate: create new event with metadata, delete old
-                if not self.read_only:
+                if self.persistence_mode is not PersistenceMode.NONE:
                     self.create_agent(session_id, agent)
                     self.memory_client.gmdp_client.delete_event(
                         memoryId=self.config.memory_id,
@@ -556,7 +554,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
         if not messages:
             return None
 
-        if self.read_only:
+        if self.persistence_mode is PersistenceMode.NONE:
             return {}
 
         is_blob = self.converter.exceeds_conversational_limit(messages[0])
@@ -875,7 +873,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
         Raises:
             SessionException: If message creation fails. On failure, messages remain in the buffer.
         """
-        if self.read_only:
+        if self.persistence_mode is PersistenceMode.NONE:
             return []
 
         with self._message_lock:
@@ -956,7 +954,7 @@ class AgentCoreMemorySessionManager(RepositorySessionManager, SessionRepository)
         Raises:
             SessionException: If agent state creation fails. On failure, agent states remain in the buffer.
         """
-        if self.read_only:
+        if self.persistence_mode is PersistenceMode.NONE:
             return []
 
         with self._agent_state_lock:

--- a/tests/bedrock_agentcore/memory/integrations/strands/test_agentcore_memory_session_manager.py
+++ b/tests/bedrock_agentcore/memory/integrations/strands/test_agentcore_memory_session_manager.py
@@ -3259,7 +3259,9 @@ class TestPersistenceMode:
         assert config.persistence_mode is PersistenceMode.FULL
 
     def test_config_accepts_none_mode(self):
-        config = AgentCoreMemoryConfig(memory_id="m", session_id="s", actor_id="a", persistence_mode=PersistenceMode.NONE)
+        config = AgentCoreMemoryConfig(
+            memory_id="m", session_id="s", actor_id="a", persistence_mode=PersistenceMode.NONE
+        )
         assert config.persistence_mode is PersistenceMode.NONE
 
     def test_config_accepts_string_value(self):
@@ -3370,7 +3372,10 @@ class TestPersistenceMode:
 
     def test_read_agent_works(self, no_persist_manager, mock_memory_client):
         mock_memory_client.list_events.return_value = [
-            {"eventId": "e1", "payload": [{"blob": '{"agent_id": "a1", "state": {}, "conversation_manager_state": {}}'}]},
+            {
+                "eventId": "e1",
+                "payload": [{"blob": '{"agent_id": "a1", "state": {}, "conversation_manager_state": {}}'}],
+            },
         ]
         result = no_persist_manager.read_agent("test-session-456", "a1")
         assert result is not None
@@ -3381,7 +3386,16 @@ class TestPersistenceMode:
             {
                 "eventId": "e1",
                 "eventTimestamp": "2024-01-01T12:00:00Z",
-                "payload": [{"conversational": {"content": {"text": '{"message": {"role": "user", "content": [{"text": "Hello"}]}, "message_id": 1}'}, "role": "USER"}}],
+                "payload": [
+                    {
+                        "conversational": {
+                            "content": {
+                                "text": '{"message": {"role": "user", "content": [{"text": "Hello"}]}, "message_id": 1}'
+                            },
+                            "role": "USER",
+                        }
+                    }
+                ],
             },
         ]
         messages = no_persist_manager.list_messages("test-session-456", "a1")
@@ -3392,7 +3406,12 @@ class TestPersistenceMode:
     def test_legacy_session_migration_skipped(self, no_persist_manager, mock_memory_client):
         mock_memory_client.list_events.side_effect = [
             [],
-            [{"eventId": "legacy-1", "payload": [{"blob": '{"session_id": "test-session-456", "session_type": "AGENT"}'}]}],
+            [
+                {
+                    "eventId": "legacy-1",
+                    "payload": [{"blob": '{"session_id": "test-session-456", "session_type": "AGENT"}'}],
+                }
+            ],
         ]
         result = no_persist_manager.read_session("test-session-456")
         assert result is not None
@@ -3402,7 +3421,12 @@ class TestPersistenceMode:
     def test_legacy_agent_migration_skipped(self, no_persist_manager, mock_memory_client):
         mock_memory_client.list_events.side_effect = [
             [],
-            [{"eventId": "legacy-1", "payload": [{"blob": '{"agent_id": "a1", "state": {}, "conversation_manager_state": {}}'}]}],
+            [
+                {
+                    "eventId": "legacy-1",
+                    "payload": [{"blob": '{"agent_id": "a1", "state": {}, "conversation_manager_state": {}}'}],
+                }
+            ],
         ]
         result = no_persist_manager.read_agent("test-session-456", "a1")
         assert result is not None

